pkg/ccl/backupccl/restore_span_covering_test.go

```diff
@@ -790,10 +802,25 @@ func TestRestoreEntryCover(t *testing.T) {
 								introducedSpanFrontier, err := createIntroducedSpanFrontier(backups, hlc.Timestamp{})
 								require.NoError(t, err)
 								cover, err := makeImportSpans(ctx, backups[numBackups-1].Spans, backups,
-									layerToIterFactory, target<<20, introducedSpanFrontier, simpleImportSpans)
+									layerToIterFactory, nil, target<<20, introducedSpanFrontier, simpleImportSpans)
 								require.NoError(t, err)
 								require.NoError(t, checkRestoreCovering(ctx, backups, backups[numBackups-1].Spans,
 									cover, target != noSpanTargetSize, execCfg.DistSQLSrv.ExternalStorage))
+
+								// Check that the correct import spans are created if the job is
+								// resumed after some random entry in the cover. The import
+								// spans created from a watermark should just be the full
+								// covering excluding entries below the watermark.
+								if len(cover) > 0 {
```

Comment:
qq: was this patched test failing before your fix in restore_span_covering.go?

Reply:
yep, both test additions are red/green tests for the fix.